### PR TITLE
kvserver: add metric and log when raft.Storage returns an error

### PR DESCRIPTION
**Pull request by sumeerbhola**
The raft.storage.error metric is incremented on an error, and the error is logged every 30s (across all replicas).

This was motivated by a test cluster that slowed to a crawl because of deliberate data loss, but was hard to diagnose. The metric could be used for alerting, since we don't expect to see transient errors.

Informs #113053

Epic: none

Release note: None

### CODE REVIEW

File: pkg/kv/kvserver/metrics.go [hunk 1]
```diff
@@ -1418,6 +1418,12 @@ cache will already have moved on to newer entries.
 		Measurement: "Bytes",
 		Unit:        metric.Unit_BYTES,
 	}
+	metaRaftStorageError = metric.Metadata{
+		Name:        "raft.storage.error",
+		Help:        "Number of calls to the raft.Storage API that returned an error",
```
Comment (erikgrinaker) [inline pkg/kv/kvserver/metrics.go:1423]: nit: the "raft.Storage API" part seems a bit too low-level for users, consider perhaps "Number of Raft storage errors".

**Review by erikgrinaker — APPROVED** [review]
LGTM, thanks for adding this!

**Review by sumeerbhola — COMMENTED** [review]
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/113245#-:-Ni-xIK1BXsLxazA23JA:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @erikgrinaker)

___
*[`pkg/kv/kvserver/metrics.go` line 1423 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/113245#gh-1376033920:-Ni-vzbM8PRQfHTU61Bx:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/0618e954842841b0121eba71d595148284acb402/pkg/kv/kvserver/metrics.go#L1423)):*
<details><summary><i>Previously, erikgrinaker (Erik Grinaker) wrote…</i></summary><blockquote>

nit: the "raft.Storage API" part seems a bit too low-level for users, consider perhaps "Number of Raft storage errors".
</blockquote></details>

Done

**Comment by sumeerbhola** [comment]
bors r=erikgrinaker

**Comment by sumeerbhola** [comment]
bors r=erikgrinaker

**Pull request merged**